File: queue/priority_queue.cpp
#include <iostream>
#define MAX 100

using namespace std;

class Node{
  public:
    int value;
    int priority;
};

class PriorityQueue{
  private:
    Node queue[MAX];
    int size;
  public:
    PriorityQueue();
    void enqueue(int, int);
    int dequeue();
    int peek();
    int getQueueValue(int);
    void display();
};

PriorityQueue::PriorityQueue(){
  Node queue[MAX];
  size=-1;
}

int PriorityQueue::peek(){
  int maxPriority = queue[0].priority;
  int index = 0;

  for (int i=1; i<=size; i++){
    if(maxPriority < queue[i].priority){
      maxPriority = queue[i].priority;
      index = i;
    }
  }

  return index;
}

int PriorityQueue::getQueueValue(int index){
  return queue[index].value;
}

void PriorityQueue::enqueue(int data, int priority){
  size++;
  queue[size].value = data;
  queue[size].priority = priority;
}

int PriorityQueue::dequeue(){
  int index = peek();

  for(int i=index; i<=size; i++){
    queue[i] = queue[i+1];
  }
  size--;

  return index;
}

void PriorityQueue::display(){
  cout << "******************" << endl;
  for (int i=0; i<=size; i++){
    cout << queue[i].value << ", " << queue[i].priority << endl;
  }
}

int main() {
  PriorityQueue pq;
  int idx;

  pq.enqueue(100, 10);
  pq.enqueue(200, 9);
  pq.enqueue(300, 10);
  pq.display();
  pq.dequeue();
  pq.display();
  
  return 0;
}

File: queue/priority_queue.py
class Node:
    def __init__(self):
        self.value = None
        self.priority = None

class PriorityQueue:
    def __init__(self, capacity):
        self.capacity = capacity
        self.size = 0
        self.queue = []

    def enqueue(self, value, priority):
        node = Node()
        node.value = value
        node.priority = priority
        self.queue.append(node)
        self.size += 1
        return True

    def peek(self):
        max_priority = self.queue[0].priority
        max_priority_index = 0

        for index in range(self.size):
            if max_priority < self.queue[index].priority:
                max_priority = self.queue[index].priority
                max_priority_index = index

        return max_priority_index

    def dequeue(self):
        index = self.peek()
        
        for i in range(index, self.size-1):
            self.queue[i] = self.queue[i+1]

        self.size -= 1
    
    def display(self):
        for i in range(self.size):
            print(f'({self.queue[i].value},{self.queue[i].priority})', end='->')
        print()

if __name__=="__main__":
    priority_queue = PriorityQueue(10)

    priority_queue.enqueue(100,10)
    priority_queue.enqueue(200,15)
    priority_queue.enqueue(300,10)
    priority_queue.enqueue(400,12)

    priority_queue.display()
    
    priority_queue.dequeue()
    priority_queue.display()
    priority_queue.dequeue()
    priority_queue.display()
    priority_queue.dequeue()
    priority_queue.display()
